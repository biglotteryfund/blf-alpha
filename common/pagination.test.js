/* eslint-env jest */
'use strict';

const { buildPagination } = require('./pagination');

describe('buildPagination', () => {
    it('should transform API pagination meta into object for views', () => {
        expect(
            buildPagination({
                total: 28,
                count: 10,
                per_page: 10,
                current_page: 1,
                total_pages: 3
            })
        ).toMatchSnapshot();

        expect(
            buildPagination({
                total: 28,
                count: 10,
                per_page: 10,
                current_page: 2,
                total_pages: 3
            })
        ).toMatchSnapshot();

        expect(
            buildPagination({
                total: 28,
                count: 10,
                per_page: 10,
                current_page: 3,
                total_pages: 3
            })
        ).toMatchSnapshot();

        // account for existing query parameters
        expect(
            buildPagination(
                {
                    total: 28,
                    count: 10,
                    per_page: 10,
                    current_page: 2,
                    total_pages: 3
                },
                { region: 'scotland' }
            )
        ).toMatchSnapshot();
    });
});
