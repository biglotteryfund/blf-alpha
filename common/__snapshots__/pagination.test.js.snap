// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildPagination should transform API pagination meta into object for views 1`] = `
Object {
  "count": 10,
  "currentPage": 1,
  "nextLink": "?page=2",
  "perPage": 10,
  "prevLink": null,
  "total": 28,
  "totalPages": 3,
}
`;

exports[`buildPagination should transform API pagination meta into object for views 2`] = `
Object {
  "count": 10,
  "currentPage": 2,
  "nextLink": "?page=3",
  "perPage": 10,
  "prevLink": "?page=1",
  "total": 28,
  "totalPages": 3,
}
`;

exports[`buildPagination should transform API pagination meta into object for views 3`] = `
Object {
  "count": 10,
  "currentPage": 3,
  "nextLink": null,
  "perPage": 10,
  "prevLink": "?page=2",
  "total": 28,
  "totalPages": 3,
}
`;

exports[`buildPagination should transform API pagination meta into object for views 4`] = `
Object {
  "count": 10,
  "currentPage": 2,
  "nextLink": "?region=scotland&page=3",
  "perPage": 10,
  "prevLink": "?region=scotland&page=1",
  "total": 28,
  "totalPages": 3,
}
`;
